Pin python dependencies to an API version, remove pyocd dependency

bridadan opened the pull request:
## Status
**READY**

## Migrations
NO

## Description
Pin all of the python dependencies to at least the latest major version to ensure stability.

Edit: Also remove pyocd as a dependency. Wherever pyocdis used, the imports are wrapped in try/except blocks and an error code returned. It seems as though it is already an optional dependency, so I'm removing the hard requirement.

## Related PRs
#157 
https://github.com/ARMmbed/icetea/pull/73
https://github.com/ARMmbed/mbed-os-tools/pull/62
https://github.com/ARMmbed/mbed-os/pull/9389

jupe commented on setup.py:
```diff
@@ -54,9 +54,9 @@ def read(fname):
-          "pyOCD!=0.13.0"
+          "six==1.*",
+          "pyserial==3.*",
+          "pyOCD>=0.13.1,<0.14.0"
```
is there coming breaking changes to pyocd?

bridadan replied:
In Mbed OS we currently pin the pyocd version because it is not pinned here and can cause breakages. I'm now being asked to update the version within Mbed OS to version 0.16.x due to new target support: https://github.com/ARMmbed/mbed-os/pull/9481#issuecomment-461118156

Do you actively use pyocd to flash? Would it be possible to make it an optional dependency like we currently do in mbed-os-tools:
- https://github.com/ARMmbed/mbed-os-tools/blob/master/setup.py#L60-L62
- https://github.com/ARMmbed/mbed-os-tools/blob/a6f73ab6353ee2d909518c99a159deaceee494d1/src/mbed_os_tools/test/host_tests_plugins/module_copy_pyocd.py#L51-L56

jupe replied:
we are not using pyocd actively yet, in fact pyocd integration is broken in master 😄. but we have near future plans to start using pyocd also in CI so this PR is just fine for now if its okay for you. We have to think about if it could be optionally but lets pin it first..

bridadan replied:
If pyocd is broken in master, do you mind if I either:

- Remove the pyocd implementation
- Make the imports conditional? So if Pyocd isn't present, it'll just print an error saying you need to install it? (we do this in mbed-os-tools, link above)

RomanSaveljev replied:
{quote}
* Remove the pyocd implementation
{quote}

This is okay with me. According to https://github.com/ARMmbed/isgtesting-arch/pull/24 we will be working on **pyOCD** integration on **mbed-flasher** branch.

bridadan commented:
I've removed pyocd from the dependency list, please let me know if you need anymore changes.

I've updated my first comment with more information on why removing the pyocd dependency without modifying the rest of the code base seems to be ok.

bridadan commented:
Anything blocking this?
